wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/hc/probe/ProbeConfig.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.workspace.server.hc.probe;
+
+/** @author Alexander Garagatyi */
+public abstract class ProbeConfig {
+  private int successThreshold = 1;
+  private int failureThreshold = 3;
+  private int timeoutSeconds = 1;
+  private int periodSeconds = 10;
+  private int initialDelaySeconds;
+
+  /**
+   * Creates probe configuration.
+   *
+   * @param successThreshold minimum consecutive successes for the probe to be considered successful
+   *     after having failed. Minimum value is 1. Defaults to 1.
```

Comment:
Doc says about default values but for me, it's not clear how to create `ProbeConfig` with default values

Reply:
Was done with config deserialization in mind. Will remove for the time being.